### DISCUSSION TITLE
Next.js app works locally, not working on Vercel

### DISCUSSION THREAD

**Discussion by intelligence** [vercel/community#6236] [Help] [answered]
### Summary

Having huge issues getting my app to work properly on Vercel. Locally everything works just fine, but when deployed it seems like the app ignores the values of variables. Never experienced this before.

```
export function ProjectMedia({ media, index }) {
  const MediaType = mediaTypes[media._type]

  console.log(media.size)
  // outputs 'small', 'medium', and 'large' on vercel and locally

  if (media.size == 'small') {
    console.log('small')
    // only output locally
  }

  if (media.size == 'medium') {
    console.log('medium')
    // only output locally
  }

  if (media.size == 'large') {
    console.log('large')
    // only output locally
  }

  return MediaType ? (
    <div className="grid grid-cols-12">
      <div
        className={clsx('flex items-center min-h-screen test', {
          'col-span-4 col-start-5': media?.size == 'small', // only applied locally
          'col-span-6 col-start-4': media?.size == 'medium', // only applied locally
          'col-span-12': media?.size == 'large', // only applied locally
          'some-class': media?.size, // applied on vercel and locally
        })}
      >
        <MediaType index={index} media={media} />
      </div>
    </div>
  ) : (
    <div>Missing media type: {media._type}</div>
  )
}
```

When I do a console.log(typeof media.size) on Vercel shows string so I'm not sure why it's doing this.

I've done multiple redeploys to clear cache, but to no avail.

### Steps to Reproduce

No error messages.

**Comment by pepiyu** [comment]
I'm having the same problem. Actually, mi app was working on Vercel till now, every new deployment ara missing css styles despite it works locally...

**Reply by pepiyu** [reply]
I solved my problem as well! I was using getInitialProps and it is not recommended since is a legacy API. Use getStaticProps instead. What was curious is, it was working locally but not on Vercel.

**Comment by intelligence** [comment, marked as answer]
This was an issue with Sanity + Stega. Invisible characters being injected in the string.
